ui/src/test/java/edu/wpi/grip/ui/codegeneration/tools/PythonPipelineInterfacer.java

```diff
@@ -0,0 +1,390 @@
+import static org.junit.Assert.fail;
+
+public class PythonPipelineInterfacer implements PipelineInterfacer {
+  StringBuilder str;
+  BufferedWriter out;
+  static File codeDir = null;
+
+  static {
+    try {
+      codeDir = new File(PipelineGenerator.class.getProtectionDomain()
+          .getCodeSource().getLocation().toURI());
+    } catch (URISyntaxException e) {
+      e.printStackTrace();
+      fail("Could not load code directory");
+    }
+  }
+
+  public PythonPipelineInterfacer(String className) {
+
+    try {
+      Runtime.getRuntime().exec("cd " + codeDir.getAbsolutePath().toString());
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    str = new StringBuilder();
+    str.append("import sys\n");
+    str.append("import cv2\n");
+    str.append("sys.path.insert(0, \'");
+    //File temp = new File("__init__.py");
+    // System.out.println(codeDir.toPath().toString());
+    str.append(codeDir.getAbsolutePath().toString());
+    str.append("\')\n");
+    str.append("import ");
+    str.append(className);
+    str.append("\n");
+    str.append("pipe = ");
+    str.append(className);
+    str.append(".");
+    str.append(className);
+    str.append("()\n");
+    try {
+      out = new BufferedWriter(new FileWriter("testing.py"));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /* (non-Javadoc)
+   * @see edu.wpi.grip.ui.codegeneration.tools.PipelineInterfacer#setSource(int, java.lang.Object)
+   */
+  @Override
+  public void setSource(int num, Object value) {
+    str.append("val = ");
+    Class type = value.getClass();
+    if (type.equals(Size.class)) {
+      Size val = (Size) value;
+      String size = "(" + val.width + "," + val.height + ")";
+      str.append(size);
+    } else if (type.equals(Point.class)) {
+      Point val = (Point) value;
+      String point = "(" + val.x + "," + val.y + ")";
+      str.append(point);
+    } else if (type.equals(Boolean.class)) {
+      if ((Boolean) value) {
+        str.append("True");
+      } else {
+        str.append("False");
+      }
+    } else {
+      str.append(value.toString());
+    }
+    str.append("\npipe.set_source");
+    str.append(num);
+    str.append("(val)\n");
+  }
+
+  /* (non-Javadoc)
+   * @see edu.wpi.grip.ui.codegeneration.tools.PipelineInterfacer#setSourceAsObject(int,  java
+   * .lang.Object)
+   */
+  @Override
+  public void setSourceAsObject(int num, Object value) {
+    setSource(num, value);
+  }
+
+  /* (non-Javadoc)
+   * @see edu.wpi.grip.ui.codegeneration.tools.PipelineInterfacer#setMatSource(int, java.io.File)
+   */
+  @Override
+  public void setMatSource(int num, File img) {
+    str.append("source = cv2.imread(\"");
+    str.append(img.getAbsolutePath());
+    str.append("\")\n");
+    str.append("pipe.set_source");
+    str.append(num);
+    str.append("(source)\n");
+  }
+
+  /* (non-Javadoc)
+   * @see edu.wpi.grip.ui.codegeneration.tools.PipelineInterfacer#process()
+   */
```

Comment:
This is an Overridden method. Javadoc is not needed -- it is automatically added from the super class.